pkgs/development/compilers/openjdk/22.nix

```diff
@@ -1,74 +1,89 @@
-{ stdenv
-, lib
-, fetchurl
-, fetchpatch
-, fetchFromGitHub
-, bash
-, pkg-config
-, autoconf
-, cpio
-, file
-, which
-, unzip
-, zip
-, perl
-, cups
-, freetype
-, alsa-lib
-, libjpeg
-, giflib
-, libpng
-, zlib
-, lcms2
-, libX11
-, libICE
-, libXrender
-, libXext
-, libXt
-, libXtst
-, libXi
-, libXinerama
-, libXcursor
-, libXrandr
-, fontconfig
-, openjdk22-bootstrap
-, ensureNewerSourcesForZipFilesHook
-, setJavaClassPath
+{
+  stdenv,
+  lib,
+  fetchurl,
+  fetchpatch,
+  fetchFromGitHub,
+  pkg-config,
+  autoconf,
+  cpio,
+  file,
+  which,
+  unzip,
+  zip,
+  perl,
+  cups,
+  freetype,
+  alsa-lib,
+  libjpeg,
+  giflib,
+  libpng,
+  zlib,
+  lcms2,
+  libX11,
+  libICE,
+  libXrender,
+  libXext,
+  libXt,
+  libXtst,
+  libXi,
+  libXinerama,
+  libXcursor,
+  libXrandr,
+  fontconfig,
+  jdk-bootstrap,
+  ensureNewerSourcesForZipFilesHook,
+  setJavaClassPath,
   # TODO(@sternenseemann): gtk3 fails to evaluate in pkgsCross.ghcjs.buildPackages
   # which should be fixable, this is a no-rebuild workaround for GHC.
-, headless ? stdenv.targetPlatform.isGhcjs
-, enableJavaFX ? false
-, openjfx
-, enableGnome2 ? true
-, gtk3
-, gnome_vfs
-, glib
-, GConf
+  headless ? stdenv.targetPlatform.isGhcjs,
+  enableJavaFX ? false,
+  openjfx,
+  enableGnome2 ? true,
+  gtk3,
+  gnome_vfs,
+  glib,
+  GConf,
+  writeShellScript,
+  testers,
 }:
 
 let
-  version = {
-    feature = "22";
-    interim = "";
-    build = "36";
-  };
+
+  # Java version format:
+  # $FEATURE.$INTERIM.$UPDATE.$PATCH
+  # See
+  # https://openjdk.org/jeps/223
+  # https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/Runtime.Version.html
+
+  featureVersion = "22";
+  info = builtins.getAttr featureVersion (builtins.fromJSON (builtins.readFile ./info.json));
```

Comment:
There should be a libs function to read json in one function

Reply:
right, there is `lib.importJSON`, updated to use that function